src/ast/passes/semantic_analyser.cpp

```diff
@@ -574,6 +574,55 @@ static const std::map<std::string, call_spec> CALL_SPEC = {
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::integer, .literal=true } } } },
+  { "strcmp",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::string } } } },
+  { "strnlen",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::integer } } } },
+  { "strlen",
```

Comment:
We already have `len` to get the number of elements in a map. I wonder if we should reuse that instead of introducing another builtin function. But again, that would require polymorphic builtin functions.

Reply:
I didn't think about it so carefully at first. You are right, we don't need to replicate kfunc 1:1, but use bpftrace's own way to call kfunc.